components/function-controller/internal/controllers/serverless/horizontalpodautoscaler.go

```diff
@@ -21,22 +21,39 @@ func (r *FunctionReconciler) isOnHorizontalPodAutoscalerChange(instance *serverl
 	}
 
 	newHpa := r.buildHorizontalPodAutoscaler(instance, deployments[0].GetName())
-	return !(len(hpas) == 1 &&
-		r.equalHorizontalPodAutoscalers(hpas[0], newHpa))
+	scalingEnabled := isScalingEnabled(instance)
+	numHpa := len(hpas)
+
+	return (scalingEnabled && numHpa != 1) ||
```

Comment:
Could you add test for that function? It's now grown big enough to have those I think